### PR TITLE
Added support for custom BLE error codes dictionary

### PR DESCRIPTION
**Pull request by mikolak**
User can add optional array of strings containing messages to be carried
by errors. This way the developer can decide what should be inside an
error message (eg. when one wants to skip MAC address from the logs).

### CODE REVIEW

File: src/BleError.js [hunk 1]
```diff
@@ -42,17 +42,17 @@ export class BleError extends Error {
    */
   reason: ?string
 
-  constructor(nativeBleError: NativeBleError | string) {
+  constructor(nativeBleError: NativeBleError | string, errorMessagesArray: Array<string>) {
```
Comment (Cierpliwy) [inline src/BleError.js]: Array is not a good match here. I think that you should create exported type, which will tel user more about how it should be constructed:

in BleError:
```js
export type BleErrorCodeMessageType = {[$Values<typeof BleErrorCode>]: string}

export const BleErrorCodeMessage: BleErrorCodeMessageType = {
```

in above function:
```js
constructor(nativeBleError: NativeBleError | string, errorMessageDict: BleErrorCodeMessageType) {
```

File: src/BleError.js [hunk 2]
```diff
@@ -66,13 +66,14 @@ export class BleError extends Error {
...
 
-export function parseBleError(errorMessage: string): BleError {
-  let bleError: BleError
+export function parseBleError(errorMessage: string, errorMessagesArray: Array<string>): BleError {
+	let bleError: BleError
+	const dictionary = errorMessagesArray ? errorMessagesArray : BleErrorCodeMessage
   try {
     const nativeBleError = JSON.parse(errorMessage)
-    bleError = new BleError(nativeBleError)
+    bleError = new BleError(nativeBleError, dictionary)
```
Comment (Cierpliwy) [inline src/BleError.js]: I think that you should use consistent names. Once you are calling it dictionary and once an array. I think that dictionary is more suitable. My Ideas: errorMessageMapping, errorMessageDict, errorMessageMappingDict

File: src/BleManager.js [hunk 3]
```diff
@@ -53,6 +54,9 @@ export class BleManager {
   // Map of active subscriptions
   _activeSubscriptions: { [id: string]: Subscription }
 
+  // Map of error codes to error messages
+  _errorMessages: Array<string>
```
Comment (Cierpliwy) [inline src/BleManager.js]: Fix type and adjust name

File: src/TypeDefinition.js [hunk 4]
```diff
@@ -66,7 +66,19 @@ export interface BleManagerOptions {
...
+	 * Optional mapping of error codes to error messages. Uses {@link BleErrorCodeMessage} 
+	 * by default.
+	 * 
+	 * To override logging UUIDs or MAC adresses in error messages copy the original object 
+	 * and overwrite values of interest to you.
+	 * 
+	 * @memberof BleManagerOptions
+	 * @instance 
+	 */
+	errorMessages?: Array<string>;
```
Comment (Cierpliwy) [inline src/TypeDefinition.js]: Wrong indent.

Comment (Cierpliwy) [reply src/TypeDefinition.js]: Also make sure to generate documentation with `npm run docs` - don't include css file in this commit.

**Comment by mikolak** [comment]
@Cierpliwy bump with review fixes

**Pull request merged**